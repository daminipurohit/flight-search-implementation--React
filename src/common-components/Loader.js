import React from 'react';
import PropTypes from 'prop-types';
import '../styles/Loader.scss';

const Loader = ({ loaderMsg = 'Loading data...' }) => (
    <div className="loader-container">
        <div className="loader-icon" />
        <div className="loader-message">{loaderMsg}</div>
    </div>
);

Loader.propTypes = {
    loaderMsg: PropTypes.string
};
export default Loader;
