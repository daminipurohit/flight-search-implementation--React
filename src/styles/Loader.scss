@import './colors.scss';

/* Safari */
@-webkit-keyframes spin {
  0% { -webkit-transform: rotate(0deg); }
  100% { -webkit-transform: rotate(360deg); }
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}

.loader-container {
    font-size: 12px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border-radius: 4px;
    padding: 10px;
    
    .loader-icon {
        border: 2px solid $concrete;
        border-radius: 50%;
        border-top: 2px solid $light-blue;
        width: 15px;
        height: 15px;
        -webkit-animation: spin 2s linear infinite; /* Safari */
        animation: spin 2s linear infinite;
        margin-bottom: 3px;  
    }
}