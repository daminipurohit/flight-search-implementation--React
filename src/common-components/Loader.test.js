import React from 'react';
import { shallow } from 'enzyme';
import Loader from './Loader';

describe('Loader', () => {
    it('should render correctly without prop', () => {
        const component = shallow(<Loader />);
        expect(component.hasClass('loader-container')).toBeTruthy();
        expect(component.find('.loader-message').text()).toBe('Loading data...');
        expect(component).toMatchSnapshot();
    });

    it('should render correctly with prop', () => {
        const component = shallow(<Loader loaderMsg="loading text added here" />);
        expect(component.hasClass('loader-container')).toBeTruthy();
        expect(component.find('.loader-message').text()).toBe('loading text added here');
        expect(component).toMatchSnapshot();
    });
});
