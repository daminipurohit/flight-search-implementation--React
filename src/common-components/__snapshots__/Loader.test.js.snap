// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render correctly with prop 1`] = `
<div
  className="loader-container"
>
  <div
    className="loader-icon"
  />
  <div
    className="loader-message"
  >
    loading text added here
  </div>
</div>
`;

exports[`Loader should render correctly without prop 1`] = `
<div
  className="loader-container"
>
  <div
    className="loader-icon"
  />
  <div
    className="loader-message"
  >
    Loading data...
  </div>
</div>
`;
